infor-design/enterprise-wc
# 1229 - Add nesting/alignment to inline-editable data grids using tree formatter

**Explain the details for making this change. What existing problem does the pull request solve?**
This PR re-works the "inline editable" style used in some IdsDataGrid cell formatters to provide a better visual display of "nested" fields alongside expand/collapse buttons.  In addition, this PR allows the "align" property provided in column config to be passed into a cell's editor, to retain the proper text alignment during editing.

**Related github/jira issue (required)**:
Closes #1229

**Steps necessary to review your pull request (required)**:
Pull/build/run, then test the following scenarios:

_New Inline Edit + Expander:_
- Open http://localhost:4300/ids-data-grid/tree-grid-inline.html
- Observe the new style in the "Id" column.  Test the different levels of nesting to ensure they make sense.
- Test expand/collapse still works
- Try editing any cell in the "Id" column.  When the editor is opened, the text should be right aligned per the column config to match the cell formatters.
- Commit new values to different levels of nesting and make sure the dirty indicator appears in the correct place.
- Use the "More Actions" menu to select different data grid row heights.  Try them all and make sure indentation/nesting/dirty indicator placement all make sense.

_Smoke test old examples:_
- Open http://localhost:4300/ids-data-grid/tree-grid-expand-collapse.html and make sure nesting/expansions appear correct on all row heights
- Open http://localhost:4300/ids-data-grid/editable-inline.html and make sure dirty indicator/row heights work on the original inline-edit style on text-formatted cells
- Open http://localhost:4300/ids-data-grid/editable.html and make sure dirty indicator/row heights work on all editable cell types.

**Included in this Pull Request**:
